modules/core/src/transitions/transition.js

```diff
@@ -37,56 +30,67 @@ export default class Transition {
    * @params props {object} - optional overriding props. see constructor
    */
   start(props) {
+    this.cancel();
+    this.settings = Object.assign({}, DEFAULT_SETTINGS, props);
+    this._inProgress = true;
+
+    const {timeline, settings} = this;
+    this._handle = timeline.addChannel({
+      delay: timeline.getTime(),
+      duration: settings.duration
+    });
+    settings.onStart(this);
+  }
+
+  /**
+   * end this transition if it is in progress.
+   */
+  end() {
     if (this._inProgress) {
-      this.onInterrupt(this);
       this.timeline.removeChannel(this._handle);
       this._handle = null;
+      this._inProgress = false;
+      this.settings.onEnd(this);
     }
-    Object.assign(this, props);
-    this._inProgress = true;
-
-    this._handle = this.timeline.addChannel({
-      delay: this.timeline.getTime(),
-      duration: this.duration
-    });
-    this.onStart(this);
   }
 
   /**
    * cancel this transition if it is in progress.
    */
   cancel() {
     if (this._inProgress) {
-      this.onInterrupt(this);
+      this.settings.onInterrupt(this);
       this.timeline.removeChannel(this._handle);
       this._handle = null;
       this._inProgress = false;
     }
   }
 
   /**
-   * update this transition.
+   * update this transition. Returns `true` if updated.
    */
   update() {
     if (!this._inProgress) {
       return false;
     }
-    const {timeline, duration, easing} = this;
-    const handle = this._handle;
 
-    let time = timeline.getTime(handle);
-    if (Number.isFinite(duration)) {
-      time = easing ? easing(time / duration) : time / duration;
-    }
-    this.time = time;
-    this.onUpdate(this);
+    this.time = this.timeline.getTime(this._handle);
+    // Call subclass method
+    this._onUpdate();
+    // Call user callback
+    this.settings.onUpdate(this);
 
-    if (timeline.isFinished(handle)) {
-      timeline.removeChannel(handle);
-      this._handle = null;
-      this._inProgress = false;
-      this.onEnd(this);
+    // This only works if `settings.duration` is set
+    // Spring transition must call `end` manually
+    if (this.timeline.isFinished(this._handle)) {
+      this.end();
```

Comment:
To me this hints that we still don't have the right abstraction, but I'm okay with this (the comment helps) until we can find a better solution.

Reply:
I agree, this is mostly for backward compatibility.